src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/links/LinkInfo.java

```diff
@@ -114,6 +115,31 @@
      */
     public abstract boolean includeTypeParameterLinks();
 
+    /**
+     * Checks whether enclosing types of inner classes should be rendered as separate
+     * links instead of generating a single link for the class with the full class name
+     * (including the name of the enclosing class) as link label.
+     *
+     * @param type the type being linked
+     * @return true if enclosing types (and their type arguments) should be rendered as
+     * separate links
+     */
+    public boolean linkEnclosingTypes(DeclaredType type) {
+        // Only render enclosing types as separate links if they have type arguments that
+        // should be linked. The rationale is that an enclosing class itself is easy to reach
+        // from the documentation of an inner class, so rendering it as separate link would
+        // add little benefit but add considerable noise, while type arguments may not
+        // otherwise be reachable from the documentation of the inner class.
```

Comment:
Do you render the _enclosing types_ as separate links, or just  the _type arguments of enclosing types_?

Even if the enclosing type has type arguments, your argument about the noise from linking the name of the enclosing type is still valid.

Reply:
Currently the first. But if someone clicks the class label in front of type arguments, thinking it would lead to the class that defines the type parameters (as in many IDEs) and instead goes to the innermost class, it'd be quite confusing too.

Reply:
Yes, it is either all type components as separate links, or none.